src/app/src/mainwindow.cpp

```diff
@@ -230,6 +235,37 @@ void MainWindow::on_ledIntensityEdit_valueChanged(double value) {
 }
 
 
+/*
+ * Check if default drive on windows has sufficient space for acquisition.
+ *
+ * @param fps setting of acquisition
+ * @param duration of acquisition
+ *
+ * @returns boolean true if space is available
+*/
+#ifdef _WIN32
+bool MainWindow::available_space_in_default_drive( double fps,double duration){
+    if(m_camera->ctx){
+        uns32 frameBytes = m_camera->ctx->frameBytes;
+        ULARGE_INTEGER  lpTotalNumberOfFreeBytes = { 0 };
+        GetDiskFreeSpaceEx(
+            m_path.c_str(),
+            nullptr,
+            nullptr,
+            & lpTotalNumberOfFreeBytes
+            );
+        std::stringstream space_string,driver_string;
+        space_string << lpTotalNumberOfFreeBytes.QuadPart;
+        driver_string << m_path;
+        spdlog::info("Drive {} has: {} bytes free for acquisition",driver_string.str(),space_string.str());
+        return lpTotalNumberOfFreeBytes.QuadPart > fps * duration * frameBytes;
+    }else{
```

Comment:
returning `false` when the camera context is invalid but showing a `Not enough space` message for the code that handles this return value could be pretty confusing for debugging potential customer issues.